src/Adapter/ContainerBuilder.php

```diff
@@ -47,26 +91,183 @@ class ContainerBuilder
      */
     public static function getContainer($name, $isDebug)
     {
-        $containerName = ucfirst($name) . 'Container';
-        $file = _PS_CACHE_DIR_ . "${containerName}.php";
+        if (null === self::$container) {
+            $builder = new ContainerBuilder();
+            self::$container = $builder->initContainer($name, $isDebug);
+        }
 
-        if (!$isDebug && file_exists($file)) {
-            require_once $file;
+        return self::$container;
+    }
 
-            return new $containerName();
+    /**
+     * @param string $name
+     * @param bool $isDebug
+     *
+     * @return ContainerInterface|SfContainerBuilder|null
```

Comment:
in which cases the Container can be null?

Reply:
never, you're right ^^